README
This is the repository of the C++ Project for the cource at CEID.
// Please translate into Greek.

This is created using vim inside a UNIX enviroment and it is build to work with the GCC compiler family.

This project is also available at https://github.com/g3orge/EcoSystem

Changelog
commit d50b68d8a912b0c533d18553bd0ba62c33ece605
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Sat Jan 14 17:22:58 2012 +0200

    Update README

commit 8f632bc4e34be411a38a5392b59ec6d40b4e1120
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Sat Jan 14 17:16:32 2012 +0200

    new README

commit 23bdf198f1a2c641a948fb1b3b37c5b21ecfdd2d
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Sat Jan 14 17:14:06 2012 +0200

    new README

commit 53e8515d1f3bcb0bd2a8d24a9bf754a4d8efa0f2
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Sat Jan 14 17:07:33 2012 +0200

    Get started with more stuff inside the sub-classes
    
    I just created the age, the size and the alias for all the basic
    organisms and improved the whole class system
    TODO: Create the map - Probaly with vector and pointers

commit b08419469199a43b85ddfca4850c5300712e0887
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Fri Jan 13 17:00:54 2012 +0200

    Implementing things into main and creating classes
    
    I just created the prototypes of all the required classes,
    and started implementing more things inside the main, the menus,
    and the map generation code.
    TODO: Add the REAL stuff into the classes
          put some organisms into the map.

commit 618af021db8126ecc0767d33538a14552c25105c
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Fri Jan 13 02:19:58 2012 +0200

    This is a whole new implementation. Basically, v2

commit d8e31562dbc83ea387b5617e48e53dfbea30c225
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Sun Jan 8 17:43:41 2012 +0200

    changes in random and wait

commit 469bc4a6cae4ae0db7ec9e1a89d82f986868c412
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 15 21:12:40 2011 +0200

    More

commit cfdd0861e2f9a3e04b644d754859814a0c805245
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 15 19:05:46 2011 +0200

    Getting started with time stuff

commit b5ae79707873c5798008b70617bb8117126a8999
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 15 15:46:44 2011 +0200

    Creating the map and placing organisms

commit 0b0ef151efd2d85e4046f0a005e21fef4e1e2c93
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 15 14:33:47 2011 +0200

    Starting to create the main function

commit e2d4325f46f51613e4b41f8c4dbe78fbd1dc389a
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Tue Dec 13 17:05:39 2011 +0200

    Implementing more stuff in the classes

commit 9f41872d2339a86c7dd696b614d4e3863540ee4f
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Tue Dec 13 15:14:50 2011 +0200

    Starting with implementing the menu and some class

commit e0e5544301e1792bd4c7ea0c2e59228f485cb780
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Tue Dec 13 14:39:38 2011 +0200

    One file

commit 13a8cf2ce4d9fe11aa2048aaabdfca1a4b1f492c
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Tue Dec 13 14:38:00 2011 +0200

    Trying to put everything in one file for starters, maybe I'll split it up later

commit 796ddaff90a132abcdf095659882bbdcce614bfa
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 1 16:00:51 2011 +0200

    Header fever and random generator

commit 15cb5f4384693cda42d91853d51c0a4729bc15a2
Merge: 77a826c 45df13f
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 1 14:59:45 2011 +0200

    Merge branch 'master' of github.com:g3orge/EcoSystem

commit 77a826c24fe628a33d0f73f9499ff38087dfddf8
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 1 14:57:31 2011 +0200

    First header file

commit 45df13f2a04cbc4155c7d9682e73233ed98e40b7
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Thu Dec 1 13:48:24 2011 +0200

    update Organisms

commit 1877a98b1551b74c1166f4ae7d452e896ab8da62
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Wed Nov 30 16:48:39 2011 +0200

    new gitignore

commit 38d9870b8eb4a78a1acf5cd45890cc1ca4607381
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Wed Nov 30 16:41:41 2011 +0200

    removing DS_Store

commit 2603b180db165bd18fe5abf531e9bdb67dc6b059
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Wed Nov 30 16:36:06 2011 +0200

    deleting useless stuff

commit f3b716387a68a6db99e6944773372b94e145149e
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Wed Nov 30 16:34:15 2011 +0200

    Original Commit

commit 2d4fafc9623f0b98a02c5a3bd34412cdce053863
Author: George Papanikolaou <g3orge.app@gmail.com>
Date:   Wed Nov 30 16:31:37 2011 +0200

    Original Commit


src/classes.h
// This is the header file for the Ecosystem project

class Organism {
    public:
        int size;       // This can be 0,1,2,3,4 - 0=DEAD
        int age;        // Number of steps they take
        int food;       // Ammount of food they need. This is 0,1,2 and when 0 their size is reducing.
        char alias;     // This is the alias for the map
        int org_eaten;  // The counter for the organisms eaten by this one
        int where_x, where_y;  // Their place in the map
        void interaction(Organism l) {
            // Plankton interaction
            if (  alias==l.alias ) {   
                if ( alias=='P') {
                    Organism L = create_Organism(5);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='Z') {
                    Organism L = create_Organism(8);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
            }
        }
        void move () {
            int movement;
            if ( where_x == 0 ) {
                if (where_y == 0 || where_y == y) {
                    movement = RandomInteger(4,6);
                }
                else {
                    movement = RandomInteger(4,8);
                }
            }
            else if (where_y == 0 || where_y == y) {
                movement = RandomInteger(2,6);
            }
            else if ( where_x == x ) {
                if (where_y == 0 || where_y == y) {
                    movement = RandomInteger(2,4);
                }
                else {
                    movement = RandomInteger(1,4);
                }
            }
            else {
                movement = RandomInteger(1,8);
            }
            // The move now:
            if (movement == 1) { interaction( map[where_x-1][where_y-1]); }
            else if (movement == 2) { interaction( map[where_x-1][where_y]); }
            else if (movement == 3) { interaction( map[where_x-1][where_y+1]); }
            else if (movement == 4) { interaction( map[where_x][where_y+1]); }
            else if (movement == 5) { interaction( map[where_x+1][where_y+1]); }
            else if (movement == 6) { interaction( map[where_x+1][where_y]); }
            else if (movement == 7) { interaction( map[where_x+1][where_y-1]); }
            else { interaction( map[where_x][where_y-1]); }
            // NO FOOD OR SIZE CHECK. The Plankton cannot die from age/size so it can provide food to the supreme organisms.
            age++;
        }
};

class NON: public Organism {  // A subclass for cells without organisms
    public:
        NON () {
            char alias = 'O';
            int size = 0;
        }
        void interaction(Organism* l) {}
        void move () {}
};

class Plankton: public Organism {
    public:
};

class Zooplankton: public Plankton {
    public:
        Zooplankton() {
            char alias = 'Z';
            int size = 1;
        }
};

class Phytoplankton: public Plankton {
    public:
        Phytoplankton() {
            char alias = 'P';
            int size = 1;
        }
};

class NPlankton: public Organism {
    public:
        void interaction(Organism l) {
            // This is the main interaction function for stuff like eating and reproduction
            if (  alias==l.alias ) {   
                if ( alias=='P') {
                    Organism L = create_Organism(5);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='Z') {
                    Organism L = create_Organism(8);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='K') {
                    Organism L = create_Organism(11);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }       
                else if ( alias=='M') {
                    Organism L = create_Organism(13);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='G') {
                    Organism L = create_Organism(15);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='T') {
                    Organism L = create_Organism(18);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='C') {
                    Organism L = create_Organism(19);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='A') {
                    Organism L = create_Organism(20);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='S') {
                    Organism L = create_Organism(22);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
                else if ( alias=='D') {
                    Organism L = create_Organism(21);                // Creating a "child" organism
                    if (  where_x == x; ||  where_y == y )        // Adding the "child" in a near cell
                        map[  where_x-1 ][  where_y-1 ] = L;
                    map[  where_x+1 ][  where_y+1 ] = L;
                }
            }
            else {
                // The Eating Option: (Only to N-Plankton)
                if (  size > l.size ) {
                    map[l.where_x][l.where_y] = this  // This Object
                    map[ where_x][ where_y] = E;      // The l organism is no more.
                    // Food code:
                    if ( size-1==l.size ) {
                        food = 3;                     // Size-depended growth and food. 
                        size++;
                    }
                    else {
                        food = 2;
                    }
                    org_eaten++;
                }
                else {
                    map[ where_x][ where_y] = l;
                    map[l.where_x][l.where_y] = E;      // The organism is no more.
                    if (l.size-1 == size) {
                        l.food = 3;
                        l.size++;
                    }
                    else {
                        l.food=2;
                    }
                    l.org_eaten++;
                }
            }
        }
        void move () {                  // The move function for NPlankton organisms with the ability to die.
            int movement;
            if ( where_x == 0 ) {
                if (where_y == 0 || where_y == y) {
                    movement = RandomInteger(4,6);
                }
                else {
                    movement = RandomInteger(4,8);
                }
            }
            else if (where_y == 0 || where_y == y) {
                movement = RandomInteger(2,6);
            }
            else if ( where_x == x ) {
                if (where_y == 0 || where_y == y) {
                    movement = RandomInteger(2,4);
                }
                else {
                    movement = RandomInteger(1,4);
                }
            }
            else {
                movement = RandomInteger(1,8);
            }
            // The move now:
            if (movement == 1) { interaction( map[where_x-1][where_y-1]); }
            else if (movement == 2) { interaction( map[where_x-1][where_y]); }
            else if (movement == 3) { interaction( map[where_x-1][where_y+1]); }
            else if (movement == 4) { interaction( map[where_x][where_y+1]); }
            else if (movement == 5) { interaction( map[where_x+1][where_y+1]); }
            else if (movement == 6) { interaction( map[where_x+1][where_y]); }
            else if (movement == 7) { interaction( map[where_x+1][where_y-1]); }
            else { interaction( map[where_x][where_y-1]); }
            // Food size check now:
            if (food=0) size--;       // Size reduced due to no food for 2 steps
            else food--;
            // Size check:
            if (size==0) {
                map[where_x][where_y] = E;      // Death.
                dead_org++;
            }
            age++;
        }
}

class Invertebrate: public NPlankton {
    public:
}

class Vertebrate: public NPlankton {
    public:
}

class Mollusca: public Invertebrate {
    public:
}

class Arthropoda: public Invertebrate {
    public:
}
// This are the real organism classes:
class Squid: public Mollusca {
    public:
        Squid() {
            char alias = 'K';
            int age = 1;
            int size = 2;
        }
}

class Octopus: public Mollusca {
    public:
        Octopus() {
            char alias = 'T';
            int age = 1;
            int size = 3;
        }
}

class Mylittus: public Mollusca {
    public:
        Mylittus() {
            char alias = 'M';
            int age = 1;
            int size = 2;
        }
}

class Crab: public Arthropoda {
    public:
        Crab() {
            char alias = 'C';
            int age = 1;
            int size = 3;
        }
}

class Shrimp: public Arthropoda {
    public:
        Shrimp() {
            char alias = 'G';
            int age = 1;
            int size = 2;
        }
}
// Higher Fish -----------------------
class Chordata: public Vertebrate {
    public:
}

class Sprattus: public Chordata {
    public:
        Sprattus() {
            char alias = 'A';
            int age = 1;
            int size = 3;
        }
}

class Dolphin: public Chordata {
    public:
        Dolphin() {
            char alias = 'D';
            int age = 1;
            int size = 4;
        }
}

class Shark: public Chordata {
    public:
        Shark() {
            char alias = 'S';
            int age = 1;
            int size = 4;
        }
}


src/eco.cpp
// This is the first file of the C++ Project 'EcoSystem', that I'm working on.
// This is the compiled file.
// © George Papanikolaou 2011

#include <iostream>
#include <cstring>
#include <ctime>
#include <vector>
#include <cstdlib>
#include "classes.h"
#include "functions.h"

using namespace std;

int RandomInteger(int low, int high);
Organism create_Organism (int digit);

int main() {
    restart_label:      // This is the required label for the restart action
    srand((int)time(0));
    NON E;              // Creating the empty object
    int x=RandomInteger(15,20);
    int y=RandomInteger(15,20);
    cout << "The map will have " << x << " length and" << y << " width" << endl;
    int N=x*y;
    cout << "Generating the system!" << endl;
    wait(1);
    Organism map[x][y];   // Trying to create a container for all the objects - The map
    // Counters:
    int step_counter = 1;
    int org_counter = 0;
    int dead_org = 0;
    cout << "Placing organisms into the map..." << endl;
    for (int i = 0; i <= x; i++) {
        for (int j = 0; j <= y; j++) {
                int digit = RandomInteger(0,20);    // Generate a random digit for the organism creation 
                Organism Y = create_Organism (digit);
                Y.where_x=i;              // Add their place inside the placeholders
                Y.where_y=j;
                map[i][j] = Y;
                if ( digit >= 4 ) org_counter++;   // Add one to the organism counter if an organism is been added
        }
    }
    // Placing a shark and a dolphin.
    bool flag=false;
    while (flag==false) {
        int place_x = RandomInteger(0,N);
        int place_y = RandomInteger(0,N);
        if (map[place_x][place_y].alias='O') {
            Shark S;
            map[place_x][place_y] = S;
            flag=true;
            org_counter++;
        }
    }
    flag=false;
    while (flag==false) {
        int place_x = RandomInteger(0,N);
        int place_y = RandomInteger(0,N);
        if (map[place_x][place_y].alias='O') {
            Dolphin D;
            map[place_x][place_y] = D;
            flag=true;
            org_counter++;
        }
    }
    cout << "Map initialized successfully" << endl;
    // Done. Starting menus and steps.
    char choice;
    info_print_label:      // <-------- This is the required label for the print and the info actions
    cout << "Press S to make a step\nP to print the map\nA to add an Organism or Infection\nR to restart\nI to get info\nAnything else to quit" << endl;
    cin >> choice;
    while (choice == 'S' || choice == 's') {
        for (int i=0;i=x;i++) {
            for (int j=0; j=y; j++) {
                map[i][j].move();           // Calling the move function of all the organisms to make a step  
            }
        }
        step_counter++;
        cout << "Press S to make a step\nP to print the map\nA to add an organism or Infection\nR to restart\nI to get info\nAnything else to quit" << endl;
        // Yes you have need to hit 's' everytime, for every step. I need something like conio.h for *nix to implement a continuous loop.
        cin >> choice;
    }
    if (choice == 'R' || choice == 'r')
        goto restart_label;
    else if (choice == 'I' || choice == 'i') {
        // Starting to count organisms:
        int P_cn=0, Z_cn=0, K_cn=0, M_cn=0, G_cn=0, T_cn=0, C_cn=0, A_cn=0, D_cn=0, S_cn=0;    // Counters
        int P_age=0, Z_age=0, K_age=0, M_age=0, G_age=0, T_age=0, C_age=0, A_age=0, D_age=0, S_age=0;    // Total age counters
        for (int i=0; i=x; i++) {
            for (int j=0; j=y; j++) {
                // Maybe there is THE NEED FOR >>*<<map[i][j] down below here
                if ( map[i][j].alias == 'P' ) {
                    P_cn++;
                    P_age = P_age+map[i][j].age;        // Creating total organism age counters for average.
                }
                else if ( map[i][j].alias == 'Z' ) {
                    Z_cn++;
                    Z_age = Z_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'K' ) {
                    K_cn++;
                    K_age = K_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'M' ) {
                    M_cn++;
                    M_age = M_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'G' ) {
                    G_cn++;
                    G_age = G_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'T' ) {
                    T_cn++;
                    T_age = T_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'C' ) {
                    C_cn++;
                    C_age = C_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'A' ) {
                    A_cn++;
                    A_age = A_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'S' ) {
                    S_cn++;
                    S_age = S_age+map[i][j].age;
                }
                else if ( map[i][j].alias == 'D' ) {
                    D_cn++;
                    D_age = D_age+map[i][j].age;
                }
        }
        cout << "There are " << org_counter << " total organisms in the map." << endl;
        cout << "The spesific organisms are:\n Phytoplankton(P) - " << P_cn << "organisms with " << P_age/P_cn << " age average."
             << "\n Zooplankton(Z) - " << Z_cn << "organisms with " << Z_age/Z_cn << " age average."
             << "\n Squid(K) - "<< K_cn << "organisms with " << K_age/K_cn << " age average."
             << "\n Mylittus(M) - " << M_cn << "organisms with " << M_age/M_cn << " age average."
             << "\n Shrimp(G) - " << G_cn << "organisms with " << G_age/G_cn << " age average."
             << "\n Octopus(T) - "<< T_cn << "organisms with " << T_age/T_cn << " age average."
             << "\n Crab(C) - " << C_cn << "organisms with " << C_age/C_cn << " age average."
             << "\n Sprattus(A) - " << A_cn << "organisms with " << A_age/A_cn << " age average."
             << "\n Dolphin(D) - " << D_cn << "organisms with " << D_age/D_cn << " age average."
             << "\n Shark(S) - " << S_cn << "organisms with " << S_age/S_cn << " age average." << endl;
        cout << "\nTotal dead organisms: " << dead_org << endl;
        cout << "System running for " << clock()/CLOCKS_PER_SEC << " seconds" << endl;
        goto info_print_label;
    }
     else if (choice == 'A' || choice == 'a') {
        // User organism creation loop:
        cout << "Press the alias of the Organism you want to add or I to add an infection" << endl;
        char adding;
        cin >> adding;
        if ( adding == 'I' || adding == 'i' ) {
            // The Infection code here:
            cout << "Press the number of the row you want the infection to appear." << endl;
            int inf_x;
            cin >> inf_x;
            cout << "Press the number of the collumn." << endl;
            int inf_y;
            cin >> inf_y;
            // Killing the near organisms:
            if ( inf_x == 0 ) {
                if (inf_y == 0 || inf_y == y) {
                    map[inf_x][inf_y] = E;     // Give the pointer to that empty object
                    map[inf_x+1][inf_y] = E;   // For the near cells ( without getting out of limits)
                    map[inf_x][inf_y+1] = E;
                    map[inf_x+1][inf_y+1] = E;
                }
                else {
                    map[inf_x][inf_y] = E;
                    map[inf_x+1][inf_y] = E;
                    map[inf_x][inf_y+1] = E;
                    map[inf_x+1][inf_y+1] = E;
                    map[inf_x][inf_y-1] = E;
                    map[inf_x+1][inf_y-1] = E;
                }
            }
            else if (inf_y == 0 || inf_y == y) {
                map[inf_x][inf_y] = E;
                map[inf_x+1][inf_y] = E;
                map[inf_x][inf_y+1] = E;
                map[inf_x+1][inf_y+1] = E;
                map[inf_x-1][inf_y+1] = E;
                map[inf_x-1][inf_y] =E;
            }
            else if ( inf_x == x ) {
                if (inf_y == 0 || inf_y == y) {
                    map[inf_x][inf_y+1] = E;
                    map[inf_x-1][inf_y+1] = E;
                    map[inf_x-1][inf_y] = E;
                }
                else {
                    map[inf_x][inf_y+1] = E;
                    map[inf_x-1][inf_y+1] = E;
                    map[inf_x-1][inf_y] = E;
                    map[inf_x-1][inf_y-1] = E;
                }
            else {
                map[inf_x][inf_y] = E;
                map[inf_x+1][inf_y] = E;
                map[inf_x][inf_y+1] = E;
                map[inf_x-1][inf_y] = E;
                map[inf_x][inf_y-1] = E;
                map[inf_x+1][inf_y+1] = E;
                map[inf_x-1][inf_y-1] = E;
                map[inf_x+1][inf_y-1] = E;
                map[inf_x-1][inf_y+1] = E;
            }
        }
        else {
            cout << "Press the number of the row you want the Organism to be added" << endl;
            int arow;
            cin >> arow;
            cout << "Now press the number of the collumn you want the Organism to be added" << endl;
            int acollumn;
            cin >> acollumn;
            int c;
            if ( adding == 'P' || adding == 'p' ) c=4;         // Matching aliases with codes for the create_Organism function
            else if ( adding == 'Z' || adding == 'z' ) c=8;
            else if ( adding == 'K' || adding == 'k' ) c=11;
            else if ( adding == 'M' || adding == 'm' ) c=13;
            else if ( adding == 'G' || adding == 'g' ) c=15;
            else if ( adding == 'T' || adding == 't' ) c=18;
            else if ( adding == 'C' || adding == 'c' ) c=19;
            else if ( adding == 'A' || adding == 'a' ) c=20;
            else if ( adding == 'D' || adding == 'd' ) c=21;
            else if ( adding == 'S' || adding == 's' ) c=22;
            else { 
                cout << "There is no " << adding << " alias in the system" << endl;
                goto info_print_label;
            }
            Organism Q = create_Organism (c);
            map[arow][acollumn] = Q;
            Q.where_x = arow;
            Q.where_y = acollumn;
            cout << "Success" << endl;
            goto info_print_label;
        }
    }
    else if (choice == 'P' || choice == 'p') {
        // Print option: (for the console)
        int i,j;
        for (i=0; i=x; i++) {
            for (j=0; j=y; j++) {
                // Maybe there is THE NEED FOR >>*<<map[i][j] down below here
                cout << map[i][j].alias << "-";
            }
            cout << "\b" << endl;   //Change the line in the stdout
        }
        cout << "-------------------------------";
        goto info_print_label;
    }
    else { cout << "System quitting after " << clock()/CLOCKS_PER_SEC << " seconds" << endl; }
    return 0;
}
